pkg/provider/provider.go

```diff
@@ -565,19 +563,6 @@ func (k *kubeProvider) Diff(
 		oldInputs.SetGroupVersionKind(gvk)
 	}
 
-	// We do not expose the "last applied config" field to the user via `Check`, so we want to ignore any diffs in the
-	// same. We achieve that by simply copying this field from the old state.
-	if oldAnnotations := oldLiveState.GetAnnotations(); oldAnnotations != nil {
-		if lastAppliedConfig, ok := oldAnnotations[lastAppliedConfigKey]; ok {
-			annotations := newInputs.GetAnnotations()
-			if annotations == nil {
-				annotations = make(map[string]string)
-			}
-			annotations[lastAppliedConfigKey] = lastAppliedConfig
-			newInputs.SetAnnotations(annotations)
-		}
-	}
```

Comment:
Is this not necessary because of this patch, or because of a previous change? If the first, why is it not needed anymore?

Reply:
This was necessary when we were calculating the patch locally using strategic or three-way JSON merge. This code was used to prevent diffs in the last-applied-config annotation. The new code does not require this because:
- In the case of dry run, the server will handle this itself
- In the case of input diffing, neither the old or new inputs contain this annotation, so adding it creates rather than prevents a diff